splunk/splunk.go

```diff
@@ -0,0 +1,14 @@
+package splunk
+
+import (
+	"fmt"
+)
+
+// CalculateAPIEndpoint builds a valid endpoint to be used against a
+// plunk Search API service
+func CalculateSearchAPIEndpoint(host, port, owner, app, outputMode string) string {
```

Comment:
```suggestion
func GetSearchAPIEndpoint(host, port, owner, app, outputMode string) string {
```

They way this is defined it would not be useful to get the value to pass into `SPLUNK_APP_API_URL` please separate between getting the general Splunk API URL, and getting the URL for the specific search endpoint.

Also I don't think `outputMode` should be hardwired into the URL like this, its cleaner to pass it via `-d` anyway.

Reply:
changed and added functions to match `fetch-uj-records`